src/opnsense/scripts/filter/lib/alias/__init__.py

```diff
@@ -181,10 +181,13 @@ def resolve(self, force=False):
                                 self._resolve_content.add(address)
                         # resolve hostnames (async) if there are any in the collected set
                         self._resolve_content = self._resolve_content.union(address_parser.resolve_dns())
-                    # Always save last recorded content to disk, also when we're not responsible for the alias
-                    # so we can use cached results when reloading a single alias.
-                    with open(self._filename_alias_content, 'w') as f_out:
-                        f_out.write('\n'.join(self._resolve_content))
+                    contents_to_write = '\n'.join(self._resolve_content)
+                    # Do not overwrite unchanged contents
+                    if undo_content != contents_to_write:
```

Comment:
Looking briefly at the code, there might be an issue when the file is not there and the new code is empty, it might be safer to ensure the file is created when it doesn't exist yet (`or not os.path.isfile(self._filename_alias_content)`).